cmd/gh-classroom/shared/shared.go

```diff
@@ -109,30 +111,53 @@ func ListAcceptedAssignments(client api.RESTClient, assignmentID int, page int,
 	return acceptedAssignmentList, nil
 }
 
+type assignmentList struct {
+	assignments []classroom.AcceptedAssignment
+	Error       error
+}
+
 func ListAllAcceptedAssignments(client api.RESTClient, assignmentID int, perPage int) (classroom.AcceptedAssignmentList, error) {
-	var page = 1
-	response, err := classroom.GetAssignmentList(client, assignmentID, page, perPage)
+
+	//Calculate the number of go channels to create. We will assign 1 channel per page
+	//so we don't put too much pressure on the API all at once
+	assignment, err := classroom.GetAssignment(client, assignmentID)
 	if err != nil {
 		return classroom.AcceptedAssignmentList{}, err
 	}
-
-	if len(response) == 0 {
-		return classroom.AcceptedAssignmentList{}, nil
-	}
-
-	//keep calling getAssignmentList until we get them all
-	var nextList []classroom.AcceptedAssignment
-	for hasNext := true; hasNext; {
-		page += 1
-		nextList, err = classroom.GetAssignmentList(client, assignmentID, page, perPage)
-		if err != nil {
-			return classroom.AcceptedAssignmentList{}, err
+	numChannels := int(math.Ceil(float64(assignment.Accepted) / float64(perPage)))
+
+	ch := make(chan assignmentList)
+	var wg sync.WaitGroup
+	wg.Add(numChannels)
+	for page := 1; page <= numChannels; page++ {
+		go func(pg int) {
+			defer wg.Done()
+			response, err := classroom.GetAssignmentList(client, assignmentID, pg, perPage)
+			ch <- assignmentList{
+				assignments: response,
+				Error:       err,
+			}
+		}(page)
+	}
+
+	go func() {
```

Comment:
```suggestion
```

I'm so sorry, this was my mistake. This should not be within a goroutine otherwise it will be non-blocking. Which is not what we want.

Reply:
No worries :), thank you for putting the time in to review it and give feedback code reviews are really fun I enjoy it!

Reply:
Pushed updates to address the issue, I put both readers and writers on their own go routines so now the wait group will block on the main thread while everything finishes up. I also added a mutex around the append call so we don't get any data races when merging in the results from each go routine:) 

Hopefully this is getting closer to a viable solution.